docs/decisions/0002-auto-enrollment-post-request-approval.rst

```diff
@@ -0,0 +1,158 @@
+2. Automatically Enrolling Learner After a Subsidy Request Approval
+===================================================================
+
+
+Status
+======
+
+?
+
+Context
+=======
+
+Learners are currently able to request a subsidy (license or coupon code) through the enterprise-access service.
+After a request is approved, the learner has to redeem the subsidy and enroll in a course manually.
+Since the goal of a learner is to access a course, we want to automatically enroll them after a request approval.
+
+For coupon code requests, requiring manual redemption is not ideal because we cannot enforce that a code is only used
+for the course it was requested for.
+Admins have some visibility into the usage of a coupon code via the admin portal. It shows only if a coupon code has been redeemed,
+not the course it's was redeemed for. We have to query ecommerce data to check the order created using the coupon code
+for that information. Automatically enrolling a learner after a request approval would guarantee that a code is used to enroll in
+the intended course.
+
+For license requests, it's less of a problem because a license can be used for all courses in an enterprise catalog and
+auto-enrollment would just save learners clicks for this case.
+
+Questions to ponder (this is a draft so I'm putting it here for visibility for now):
+* Does enrolling a learner after a request approval provide a good/better user experience?
```

Comment:
I had not thought of this but we I would like to draw Joe's attention to this one.